client/buffered_client.go

```diff
@@ -0,0 +1,131 @@
+package client
+
+import "time"
+
+// BufferConfig is used to specify how frequently a BufferedClient should flush its dataset to the influxdb server.
+// Database: The Database to write points to (gets passed through to BatchPoints).
+// FlushMaxPoints: Buffer at most this many points in memory before flushing to the server.
+// FlushMaxWaitTime: Buffer points in memory for at most this amount of time before flushing to the server.
+type BufferConfig struct {
+	Database         string
+	FlushMaxPoints   int
+	FlushMaxWaitTime time.Duration
+}
+
+// NewBufferedClient will instantiate and return a connected BufferedClient.
+func NewBufferedClient(clientConfig Config, bufferConfig BufferConfig) (bufferedClient *BufferedClient, err error) {
+	client, err := NewClient(clientConfig)
+	if err != nil {
+		return
+	}
+	bufferedClient = &BufferedClient{
+		Client:       client,
+		bufferConfig: bufferConfig,
+		ingestChan:   make(chan Point, bufferConfig.FlushMaxPoints/3),
+		closeChan:    make(chan chan bool, 1),
+		flushTimer:   time.NewTimer(bufferConfig.FlushMaxWaitTime),
+		pointsBuf:    make([]Point, bufferConfig.FlushMaxPoints),
+		pointsIndex:  0,
+	}
+	go bufferedClient.ingestAndFlushLoop()
+	return
+}
+
+// BufferedClient is used to buffer points in memory and periodically flush them to the server
+type BufferedClient struct {
+	*Client
+	bufferConfig BufferConfig
+	ingestChan   chan Point
+	closeChan    chan chan bool
+	flushTimer   *time.Timer
+	pointsBuf    []Point
+	pointsIndex  int
+}
+
+// Add a Point with the given values to the BufferedClient.
+// If the BufferedClient is closed, didAdd is false
+func (b *BufferedClient) Add(measurement string, val interface{}, tags map[string]string, fields map[string]interface{}) (didAdd bool) {
+	ingestChan := b.ingestChan
+	if ingestChan == nil {
+		return
+	}
+	if fields == nil {
+		fields = make(map[string]interface{}, 1)
+	}
+	fields["value"] = val
+	ingestChan <- Point{
+		Measurement: measurement,
+		Tags:        tags,
+		Fields:      fields,
+		Time:        time.Now(),
+	}
+	didAdd = true
+	return
+}
+
+// Close will close the BufferedClient. While closing, it will flush any points from Add()
+// This method executes asynchronously, but it returns a channel which can be read from to ensure that the buffered client
+// Once the client
+func (b *BufferedClient) Close() (didCloseChan chan bool) {
```

Comment:
`Close` should follow this interface: http://golang.org/pkg/io/#Closer

Basically, it should block until closed and then return an error, if any.